ts/features/cieLogin/components/CieIdLoginWebView.tsx

```diff
@@ -0,0 +1,169 @@
+
+export type WebViewLoginNavigationProps = {
+  spidLevel: SpidLevel;
+  isUat: boolean;
+};
+
+const iOSUserAgent =
+  "Mozilla/5.0 (iPhone; CPU iPhone OS 14_0_1 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.0 Mobile/15E148 Safari/604.1";
+const defaultUserAgent = Platform.select({
+  ios: iOSUserAgent,
+  default: undefined
+});
+
+const originSchemasWhiteList = ["https://*", "iologin://*"];
+
+export type CieIdLoginProps = {
+  spidLevel: SpidLevel;
+  isUat: boolean;
+};
+
+const CieIdLoginWebView = ({ spidLevel, isUat }: CieIdLoginProps) => {
+  const navigation = useIONavigation();
+  const webView = useRef<WebView>(null);
+  const dispatch = useIODispatch();
+  const [authenticatedUrl, setAuthenticatedUrl] = useState<string | null>(null);
+  const loggedInAuth = useIOSelector(loggedInAuthSelector, _isEqual);
+  const loginUri = getCieIDLoginUri(spidLevel, isUat);
+
+  const navigateToCieIdAuthenticationError = useCallback(() => {
+    navigation.navigate(ROUTES.AUTHENTICATION, {
+      screen: ROUTES.AUTHENTICATION_CIE_ID_ERROR
+    });
+  }, [navigation]);
+
+  const { shouldBlockUrlNavigationWhileCheckingLollipop, webviewSource } =
+    useLollipopLoginSource(navigateToCieIdAuthenticationError, loginUri);
+
+  useEffect(() => {
+    // https://reactnative.dev/docs/linking#open-links-and-deep-links-universal-links
+    const urlListenerSubscription = Linking.addEventListener(
+      "url",
+      ({ url }) => {
+        // if the url is of this format: iologincie:https://idserver.servizicie.interno.gov.it/idp/login/livello2mobile?value=e1s2
+        // extract the part after iologincie: and dispatch the action to handle the login
+        if (url.startsWith("iologincie:")) {
+          const [, continueUrl] = url.split("iologincie:");
+
+          if (continueUrl) {
+            // https://idserver.servizicie.interno.gov.it/cieiderror?cieid_error_message=Operazione_annullata_dall'utente
+            // We check if the continueUrl is an error
+            if (continueUrl.indexOf("cieiderror") !== -1) {
+              // And we extract the error message and show it in an alert
+              const [, errorMessage] = continueUrl.split(
+                "cieid_error_message="
+              );
+              // TODO: remove this after https://pagopa.atlassian.net/browse/IOPID-2322
+              Alert.alert("Login error ❌", errorMessage ?? "error");
+            } else {
+              setAuthenticatedUrl(continueUrl);
+            }
+          }
+        }
+      }
+    );
+
+    return () => urlListenerSubscription.remove();
+  }, []);
+
+  const handleOnShouldStartLoadWithRequest = (
+    event: WebViewNavigation
+    // eslint-disable-next-line sonarjs/cognitive-complexity
+  ): boolean => {
+    const url = event.url;
+
+    if (shouldBlockUrlNavigationWhileCheckingLollipop(url)) {
+      return false;
+    }
+
+    if (url.indexOf("token=") !== -1) {
+      const [, token] = url.split("token=");
```

Comment:
```suggestion
    if (url.indexOf("profile.html?token=") !== -1) {
      const [, token] = url.split("profile.html?token=");
```

Here it'd be better to check also the existence of `profile.html` ad done for the other IdP L2 and EIC L3.
The session token from IO-BE comes always with this html page. What do you think?

And what about adding that value to a constant?

Reply:
I agree, thank you for this suggestions, I will apply them!